GAE/src/org/akvo/flow/rest/DataPointAssignmentRestService.java

```diff
@@ -0,0 +1,181 @@
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.rest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.akvo.flow.dao.DataPointAssignmentDao;
+import org.akvo.flow.domain.persistent.DataPointAssignment;
+import org.akvo.flow.rest.dto.DataPointAssignmentDto;
+import org.akvo.flow.rest.dto.DataPointAssignmentPayload;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.waterforpeople.mapping.app.web.rest.ResourceNotFoundException;
+import org.waterforpeople.mapping.app.web.rest.dto.RestStatusDto;
+
+import com.gallatinsystems.common.Constants;
+import com.google.appengine.api.datastore.KeyFactory;
+
+
+@Controller
+@RequestMapping("/data_point_assignments")
+public class DataPointAssignmentRestService {
+
+    private DataPointAssignmentDao dataPointAssignmentDao = new DataPointAssignmentDao();
+
+    @RequestMapping(method = RequestMethod.GET, value = "")
+    @ResponseBody
+    public Map<String, List<DataPointAssignmentDto>> listSomeOrAll(
+            @RequestParam(value = "surveyAssignmentId", required = false) Long surveyAssignmentId,
+            @RequestParam(value = "deviceId", required = false) Long deviceId
+        ) {
+        final HashMap<String, List<DataPointAssignmentDto>> response = new HashMap<String, List<DataPointAssignmentDto>>();
+        final List<DataPointAssignmentDto> results = new ArrayList<DataPointAssignmentDto>();
+
+        if (surveyAssignmentId != null) {
+            for (DataPointAssignment dpa : dataPointAssignmentDao.listBySurveyAssignment(surveyAssignmentId)) {
+                results.add(marshallToDto(dpa));
+            }
+        } else if (deviceId != null) {
+            for (DataPointAssignment dpa : dataPointAssignmentDao.listByDevice(deviceId)) {
+                results.add(marshallToDto(dpa));
+            }
+        } else { //All of them
+            for (DataPointAssignment dpa : dataPointAssignmentDao.list(Constants.ALL_RESULTS)) {
```

Comment:
Do we actually ever have a use case where we need to list all of them?

Reply:
Not outside of testing/debugging right now, but it is nicely orthogonal w the other REST services.